sktime/networks/scinet.py

```diff
@@ -0,0 +1,802 @@
+"""Deep Learning Forecasters using SCINet Forecaster."""
+
+import math
+
+import numpy as np
+
+from sktime.utils.dependencies import _check_soft_dependencies
+
+if _check_soft_dependencies("torch", severity="none"):
+    import torch
+    import torch.nn as nn
+    import torch.nn.functional as F
+    from torch.autograd import Variable
+
+    nn_module = nn.Module
+else:
+
+    class nn_module:
+        """Dummy class if torch is unavailable."""
+
+
+class Splitting(nn_module):
+    """
+    A neural network module for splitting input into even and odd parts.
+
+    The `Splitting` class defines a PyTorch module that splits an input
+    tensor into two parts: the even-indexed and odd-indexed elements.
+    This can be used in various tasks requiring input splitting or masking.
+    """
+
+    def __init__(self):
+        """
+        Initialize the Splitting module.
+
+        This function initializes the `Splitting` class by calling the
+        `__init__` method of its superclass (`nn.Module`). No additional
+        parameters are needed for initialization.
+        """
+        super().__init__()
```

Comment:
other models that use pytorch adapter, have network classes wrapped inside another class. I don't remember what was the purpose of that. @fkiraly and @benHeid if that was just trivial, we can keep the network classes here as they are.

Reply:
the original purpose was to add above neural network classes:

* tagging
* parameter management
* isolation of `torch`
* independence from backend, e.g., `torch` vs `tensorflow`

Although we may want to revisit them, as they never quite reached an abstract/modular level as one would expect from this kind of class.

Reply:
isn't `torch` already isolated in this case? we are using dummy class for `nn_module` and no `torch` import happens in `__init__`,
so because of the other 3 reasons, you are suggesting we keep these wrapped inside another class?

Reply:
@Sohaib-Ahmed21 Thanks for updating the network classes